cli/cmd/upgrade.go

```diff
@@ -42,52 +42,58 @@ and apply any required migrations and adaptations.`,
 			os.Exit(1)
 		}
 
-		cm, err := client.CoreV1().ConfigMaps(ns).Get(ctx, resources.OdigosDeploymentConfigMapName, metav1.GetOptions{})
-		if err != nil {
-			fmt.Println("Odigos upgrade failed - unable to read the current Odigos version for migration")
-			os.Exit(1)
-		}
+		var operation string
 
-		currOdigosVersion := cm.Data["ODIGOS_VERSION"]
-		if currOdigosVersion == "" {
-			fmt.Println("Odigos upgrade failed - unable to read the current Odigos version for migration")
-			os.Exit(1)
-		}
+		if !cmd.Flag("skip-version-check").Changed {
```

Comment:
the change here is adding this new flag since the current upgrade behavior requires a valid formatted version `v1.2.3` while the tests use the tag `e2e-test`. this flag allows the test to bypass the checks